docs/vcl.md

```diff
@@ -0,0 +1,80 @@
+## Introduction and Overview
+
+VCL is a communication library in VPP that provides a socket-based interface for client/server communication with VPP. It is designed to enable client applications to establish connections with VPP, send and receive control messages, and perform configuration and management operations.
+
+## VCL feature in CalicoVPP
+
+### Enabling VCL feature in CalicoVPP
+
+To enable vcl in your calicoVPP cluster, make sure parameter is set here:
+
+```yaml
+# dedicated configmap for VPP settings
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: calico-vpp-config
+  namespace: calico-vpp-dataplane
+data:
+  CALICOVPP_FEATURE_GATES: |-
+  {
+    "vclEnabled": true
+  }
+```
+
+### Using VCL in pods:
+
+To enable VCL on a pod, use the following annotation:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: samplepod
+  annotations:
+    "cni.projectcalico.org/vppVcl": "enable"
+```
+This will trigger vpp to create the VCL socket and add a session, which is an abstraction that allows applications to send and receive data over the network.
+
+### Sockets
+
+VCL in VPP uses a socketfile: a Unix domain socket used for communication between endpoints. This allows sessions to initiate connections.
+We use an abstract socket for our interface. This socket address is usually represented as an abstract name beginning with a null byte followed by the desired name, typically "@mysocket".
+VPP api has a particular syntax for abstract sockets: using the keyword `abstract:` then `netns_name:`. 
+For example: `abstract:vpp/session,netns_name=/var/run/netns/cni-75e26661-4119-90a4-b9d2-0b00489f76f3`. This syntax appears in vppctl.
+
+## Troubleshooting vcl interface creation
+
+To check abstract socket creation in linux, you can run the following command on the pod running vcl:
+
+```bash
+lsof -U | grep session
+vpp_main 3046023 root   33u  unix 0xffff9eaeb1943740      0t0 396363922 @vpp/session type=SEQPACKET
+```
+
+To check sessions in vpp cli:
+
+```bash
+    _______    _        _   _____  ___ 
+ __/ __/ _ \  (_)__    | | / / _ \/ _ \
+ _/ _// // / / / _ \   | |/ / ___/ ___/
+ /_/ /____(_)_/\___/   |___/_/  /_/    
+
+vpp# sh session verbose
+Connection                                                  State          Rx-f      Tx-f      
+[0:0][CT:T] 0.0.0.0:5201->0.0.0.0:0                         LISTEN         0         0         
+[0:1][T] 11.0.0.139:5201->0.0.0.0:0                         LISTEN         0         0         
+[0:2][CT:T] 0.0.0.0:5201->0.0.0.0:0                         LISTEN         0         0         
+[0:3][T] 11.0.0.140:5201->0.0.0.0:0                         LISTEN         0         0         
+Thread 0: active sessions 4
+```
+
+## Testing VCL feature
+
+VCL can be tested using iperf network performance tool.
+here is a [deployment](../test/yaml/iperf3-vcl/test.yaml) example with client/server pods for tests.
+```bash
+test/scripts/vppdev.sh tst up iperf3-vcl
+test/scripts/vppdev.sh cases ipv4_vcl
+
+```
```

Comment:
We should probably directly embed the yamls (as we already have a part of it above)
and add a section about what the entrypoint does in the iperf3-vcl case (the `iperf3-vcl.sh` script),
i.e. :
- generate the `/etc/vpp/vcl.conf` configuration file which will be picked up by the VCL library as we did set `VCL_CONFIG=/etc/vpp/vcl.conf`
- use `LD_PRELOAD=/usr/local/lib/vpp/libvcl_ldpreload.so` which is roughly equivalent of dynamically linking the iperf3 binary against the VCL instead of the libc (so piping the connect() and accept() syscall to VPP instead of linux)
- and then doing `iperf3 -4 -s` which creates the listener session
````
[0:1][T] 11.0.0.139:5201->0.0.0.0:0 LISTEN
````

Reply:
This session is probably also better put before the Troubleshooting one